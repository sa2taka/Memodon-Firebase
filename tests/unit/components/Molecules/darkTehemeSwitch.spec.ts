import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import DarkThemeSwitch from '@/components/Molecules/darkThemeSwitch.vue';

const localVue = createLocalVue();

localVue.use(Vuetify);

describe('Component', () => {
  let wrapper: any;

  beforeEach(() => {
    const store = new Vuex.Store({
      state: {
        theme: {
          themes: 'light',
        },
      },
    });
    wrapper = shallowMount(DarkThemeSwitch, { store, localVue });
  });

  test('is a Vue instance', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('become dark mode', () => {
    const functionMock = jest.fn();
    wrapper.setMethods({
      onChangeTheme: functionMock,
    });
    wrapper.vm.$data.isDark = true;
  });
});
