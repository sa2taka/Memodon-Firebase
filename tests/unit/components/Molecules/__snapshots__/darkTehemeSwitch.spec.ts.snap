// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is a Vue instance 1`] = `
<v-switch-stub
  backgroundcolor=""
  class="center-switch"
  color="accent"
  errorcount="1"
  errormessages=""
  id="dark-mode-switch"
  messages=""
  ripple="true"
  rules=""
  successmessages=""
  valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
/>
`;
